dexs/equalizer-exchange/index.ts

```diff
@@ -0,0 +1,218 @@
+
+  const poolLength = (await sdk.api.abi.call({
+    target: FACTORY_ADDRESS,
+    chain: 'fantom',
+    abi: ABIs.allPairsLength,
+  })).output;
+
+  const poolsRes = await sdk.api.abi.multiCall({
+    abi: ABIs.allPairs,
+    calls: Array.from(Array(Number(poolLength)).keys()).map((i) => ({
+      target: FACTORY_ADDRESS,
+      params: i,
+    })),
+    chain: 'fantom'
+  });
+
+  const lpTokens = poolsRes.output
+    .map(({ output }) => output);
+
+  const [underlyingToken0, underlyingToken1] = await Promise.all(
+    ['token0', 'token1'].map((method) =>
+      sdk.api.abi.multiCall({
+        abi: PAIR_TOKEN_ABI(method),
+        calls: lpTokens.map((address) => ({
+          target: address,
+        })),
+        chain: 'fantom'
+      })
+    )
+  );
+
+  const tokens0 = underlyingToken0.output.map((res) => res.output);
+  const tokens1 = underlyingToken1.output.map((res) => res.output);
+  const pairInfos = await Promise.all(lpTokens.map((_, index: number) => getPairInfo(lpTokens[index], [tokens0[index], tokens1[index]])));
+  const todaysBlock = (await getBlock(todaysTimestamp, 'fantom', {}));
+  const yesterdaysBlock = (await getBlock(yesterdaysTimestamp, 'fantom', {}));
+  const logs: ILog[][] = (await Promise.all(lpTokens.map((address: string) => sdk.api.util.getLogs({
+      target: address,
+      topic: topic_name,
+      toBlock: yesterdaysBlock,
+      fromBlock: todaysBlock,
+      keys: [],
+      chain: 'fantom',
+      topics: [topic0]
+  }))))
+    .map((p: any) => p)
+    .map((a: any) => a.output);
+  const coins = [...tokens0, ...tokens1].map((e: string) => `fantom:${e}`);
+  const prices = await getPrices(coins, timestamp);
+  const untrackVolumes: number[] = lpTokens.map((_: string, index: number) => {
+    const log: IAmount[] = logs[index]
+    .map((e:ILog)  => {return  { ...e, data: e.data.replace('0x', '') }})
+    .map((p: ILog) => {
+      const amount0In = new BigNumber('0x'+p.data.slice(0, 64)).toString();
+      const amount1In = new BigNumber('0x'+p.data.slice(64, 128)).toString();
+      const amount0Out = new BigNumber('0x'+p.data.slice(128, 192)).toString();
+      const amount1Out = new BigNumber('0x'+p.data.slice(192, 256)).toString();
+      return {
+        amount0In,
+        amount1In,
+        amount0Out,
+        amount1Out,
+      } as IAmount
+    }) as IAmount [];
+    const token0Price = (prices[`fantom:${tokens0[index].toLowerCase()}`]?.price || 0);
+    const token1Price = (prices[`fantom:${tokens1[index].toLowerCase()}`]?.price || 0);
+    const totalAmount0 = log
+      .reduce((a: number, b: IAmount) => Number(b.amount0In)+ Number(b.amount0Out) + a, 0) / 10 ** pairInfos[index].token0.decimals * token0Price;
+    const totalAmount1 = log
+      .reduce((a: number, b: IAmount) => Number(b.amount1In)+ Number(b.amount1Out) + a, 0) / 10 ** pairInfos[index].token1.decimals * token1Price;
+    const untrackAmountUSD = (totalAmount0 + totalAmount1) / 2;
```

Comment:
why count it twice and then divide by 2?

Reply:
would be better if you just counted the token for which we have price data i think?

Reply:
idk, i follow code base from subgraph uniswap they div by 2
<img width="920" alt="Screenshot 2566-01-13 at 19 24 02" src="https://user-images.githubusercontent.com/11234402/212319632-1298e5f9-0922-40e3-9a34-6a32937b8c7f.png">